skbio/sequence/_iupac_sequence.py

```diff
@@ -164,23 +167,71 @@ def _motifs(self):
 
     @overrides(Sequence)
     def __init__(self, sequence, metadata=None, positional_metadata=None,
-                 validate=True, case_insensitive=False):
+                 validate=True, lowercase=False):
         super(IUPACSequence, self).__init__(
             sequence, metadata, positional_metadata)
 
-        if case_insensitive:
-            self._convert_to_uppercase()
+        # Note we are checking identity against the False object here and not
+        # just falsiness
+        if lowercase is not False:
+            lowercase_mask = self._bytes > self._ascii_lowercase_boundary
+            self._convert_to_uppercase(lowercase_mask)
+            if not isinstance(lowercase, bool):
```

Comment:
I think it should be limited to `six.string_types` that way the same key can be used for the `lowercase` method. Without this `lowercase=` would behave differently between `io` and `IUPACSequence`